packages/amplify-e2e-tests/src/__tests__/rds-pg-relational-directives.test.ts

```diff
@@ -0,0 +1,821 @@
+      'CREATE TABLE profile (id VARCHAR(40) PRIMARY KEY, details VARCHAR(255), userId VARCHAR(40))',
+      'CREATE TABLE zipcode (zip VARCHAR(40) PRIMARY KEY, city VARCHAR(255), state VARCHAR(255), country VARCHAR(255))',
+      "INSERT INTO zipcode VALUES ('20158', 'Hamilton', 'VA', 'US')",
+      "INSERT INTO zipcode VALUES ('20160', 'Lincoln', 'VA', 'US')",
+    ];
+
+    const db = await setupRDSInstanceAndData(dbConfig, queries);
+    port = db.port;
+    host = db.endpoint;
+  };
+
+  const cleanupDatabase = async (): Promise<void> => {
+    await deleteDBInstance(identifier, region);
+  };
+
+  const createSqlStatementsDirectory = (apiName: string): void => {
+    const sqlStatementsDirPath = path.join(projRoot, 'amplify', 'backend', 'api', apiName, 'sql-statements');
+    if (!existsSync(sqlStatementsDirPath)) {
+      mkdirSync(sqlStatementsDirPath);
+    }
+  };
+
+  const createSqlStatementFile = (apiName: string, statement: string, fileName: string): void => {
+    const sqlStatementsDirPath = path.join(projRoot, 'amplify', 'backend', 'api', apiName, 'sql-statements');
+    const scriptFilePath = path.join(sqlStatementsDirPath, `${fileName}.sql`);
+    writeFileSync(scriptFilePath, statement);
+  };
+
+  const initProjectAndImportSchema = async (): Promise<void> => {
+    const apiName = 'rdsrelationalapi';
+    await initJSProjectWithProfile(projRoot, {
+      disableAmplifyAppCreation: false,
+      name: projName,
+    });
+
+    const metaAfterInit = getProjectMeta(projRoot);
+    region = metaAfterInit.providers.awscloudformation.Region;
+    await setupDatabase();
+
+    const rdsSchemaFilePath = path.join(projRoot, 'amplify', 'backend', 'api', apiName, 'schema.rds.graphql');
+
+    await addApiWithoutSchema(projRoot, { transformerVersion: 2, apiName });
+
+    await importRDSDatabase(projRoot, {
+      database,
+      engine: 'postgres',
+      host,
+      port,
+      username,
+      password,
+      useVpc: true,
+      apiExists: true,
+    });
+
+    const schema = /* GraphQL */ `
+      input AMPLIFY {
+        engine: String = "postgres"
+        globalAuthRule: AuthRule = { allow: public }
+      }
+      type Blog @model @refersTo(name: "blog") {
+        id: String! @primaryKey
+        content: String
+        posts: [Post] @hasMany(references: ["blogId"])
+      }
+      type Post @model @refersTo(name: "post") {
+        id: String! @primaryKey
+        content: String
+        blogId: String! @refersTo(name: "blogid")
+        blog: Blog @belongsTo(references: ["blogId"])
+      }
+      type User @model @refersTo(name: "user") {
+        id: String! @primaryKey
+        name: String
+        profile: Profile @hasOne(references: ["userId"])
+      }
+      type Profile @model @refersTo(name: "profile") {
+        id: String! @primaryKey
+        details: String
+        userId: String! @refersTo(name: "userid")
+        user: User @belongsTo(references: ["userId"])
+      }
```

Comment:
For the fields and model names that have `refersTo`, we wouldn't need them if we put these in quotes when creating the tables. Then essentially we can have the tests shared between MySQL and Postgres and put it in the folder `rds-v2-tests-common` that I added in my [other PR](https://github.com/aws-amplify/amplify-category-api/pull/2005/files).